app/questionnaire/questionnaire_schema.py

```diff
@@ -94,11 +95,37 @@ def __init__(
         self._populate_answer_dependencies()
         self._populate_when_rules_section_dependencies()
         self._populate_calculated_summary_section_dependencies()
+        self._populate_min_max_map()
 
     @cached_property
     def answer_dependencies(self) -> ImmutableDict[str, set[AnswerDependent]]:
         return ImmutableDict(self._answer_dependencies_map)
 
+    def _append_to_min_max_map(
+        self, min_max: Literal["minimum", "maximum"], answer_id: str, answers: list
+    ) -> None:
+        if value := answers[0].get(min_max, {}).get("value"):
+            if isinstance(value, int):
+                self.min_and_max_map[answer_id] = str(value)
+            elif isinstance(value, dict) and value:
+                if (
+                    value.get("source") == "answers"
+                    and value["identifier"] in self.min_and_max_map
+                ):
+                    self.min_and_max_map[answer_id] = self.min_and_max_map[
+                        value["identifier"]
+                    ]
+
+    def _populate_min_max_map(self) -> None:
+        for answer_id, answers in self._answers_by_id.items():
+            if (answer_type := answers[0].get("type")) and answer_type not in [
+                "Date",
+                "MonthYearDate",
+                "YearDate",
+            ]:
```

Comment:
Why these types?. Explicitly check types that allow min and max?. This would check things like Checkbox etc.